frontend/src/views/Viewlevelstable.vue

```diff
@@ -0,0 +1,71 @@
+<template>
+  <div class="PageWrapper">
+    <div class="TableContainer">
+      <h2>Attached Levels for Role ID: {{ roleId }}</h2>
+      <br />
+      <ElTable :data="levels" style="width: 100%">
+        <ElTableColumn prop="name" label="Levels" />
+      </ElTable>
+      <div class="NewLevel">
+        <ElButton type="primary" @click="redirectToLevelView">Create Level</ElButton>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted, watch } from 'vue'
+import { ElTable, ElTableColumn, ElButton } from 'element-plus'
+import { useRouter, useRoute } from 'vue-router'
+import { getLevels } from '@/stores/levels/getters'
+import { useLevelsStore } from '@/stores/levels'
+
+const router = useRouter()
+const route = useRoute()
+
+const roleId = ref(route.params.roleId?.toString() || '')
```

Comment:
I don't think you need this ref. If you get this value inside the fetch function, is already enough